create-rust-app/src/mailer.rs

```diff
@@ -1,8 +1,32 @@
+#[cfg(feature = "plugin_auth")]
+use crate::auth::mail::{
+    auth_activated, auth_password_changed, auth_password_reset, auth_recover_existent_account,
+    auth_recover_nonexistent_account, auth_register,
+};
+#[cfg(feature = "plugin_auth")]
+use dyn_clone::{clone_trait_object, DynClone};
+
 use lettre::message::{Message, MultiPart};
 use lettre::transport::smtp::authentication::Credentials;
 use lettre::transport::stub::StubTransport;
 use lettre::{SmtpTransport, Transport};
 
+// the DyncClone trait bound is for cloning, and the
+// Send trait bound is for thread-safety
+#[cfg(feature = "plugin_auth")]
+/// A trait that defines the behavior of an email template
+pub trait EmailTemplates: DynClone + Send {
+    fn send_activated(&self, mailer: &Mailer, to_email: &str);
+    fn send_password_changed(&self, mailer: &Mailer, to_email: &str);
+    fn send_password_reset(&self, mailer: &Mailer, to_email: &str);
+    fn send_recover_existent_account(&self, mailer: &Mailer, to_email: &str, link: &str);
+    fn send_recover_nonexistent_account(&self, mailer: &Mailer, to_email: &str, link: &str);
+    fn send_register(&self, mailer: &Mailer, to_email: &str, link: &str);
+}
+
+#[cfg(feature = "plugin_auth")]
+clone_trait_object!(EmailTemplates);
```

Comment:
allow `Box<dyn EmailTemplate>` to be cloned (thereby allowing all the structs that have members of that type to be cloned as well (specifically the Mailer and AppData structs))